src/PlusDataCollection/Atracsys/vtkPlusAtracsysTracker.cxx

```diff
@@ -0,0 +1,396 @@
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusAtracsysTracker::InternalDisconnect()
+{
+  LOG_TRACE("vtkPlusAtracsysTracker::InternalDisconnect");
+  this->Internal->Tracker->EnableUserLED(false);
+
+  ATRACSYS_RESULT result;
+  if ((result = this->Internal->Tracker->Disconnect()) != ATR_SUCCESS)
+  {
+    LOG_ERROR(this->Internal->Tracker->ResultToString(result));
+    return PLUS_FAIL;
+  }
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusAtracsysTracker::InternalStartRecording()
+{
+  LOG_TRACE("vtkPlusAtracsysTracker::InternalStartRecording");
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusAtracsysTracker::InternalStopRecording()
+{
+  LOG_TRACE("vtkPlusAtracsysTracker::InternalStopRecording");
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusAtracsysTracker::InternalUpdate()
+{
+  LOG_TRACE("vtkPlusAtracsysTracker::InternalUpdate");
+  const double unfilteredTimestamp = vtkPlusAccurateTimer::GetSystemTime();
+
+  std::vector<AtracsysTracker::Marker> markers;
+
+  ATRACSYS_RESULT result = result = this->Internal->Tracker->GetMarkersInFrame(markers);
+  if (result != ATR_SUCCESS && result != AtracsysTracker::ATRACSYS_RESULT::ERROR_NO_FRAME_AVAILABLE)
+  {
+    LOG_ERROR(this->Internal->Tracker->ResultToString(result));
+    return PLUS_FAIL;
+  }
+  else if (result == AtracsysTracker::ATRACSYS_RESULT::ERROR_NO_FRAME_AVAILABLE)
+  {
+    // waiting for frame
+    return PLUS_SUCCESS;
+  }
+
+  std::map<int, std::string>::iterator it;
+  for (it = begin(this->Internal->FtkGeometryIdMappedToToolId); it != end(this->Internal->FtkGeometryIdMappedToToolId); it++)
+  {
+    bool toolUpdated = false;
+
+    std::vector<AtracsysTracker::Marker>::iterator mit;
+    for (mit = begin(markers); mit != end(markers); mit++)
+    {
+      if (it->first != mit->GetGeometryID())
+      {
+        continue;
+      }
+      // check if tool marker registration falls above maximum
+      if (mit->GetFiducialRegistrationErrorMm() > this->Internal->MaxMeanRegistrationErrorMm)
+      {
+        LOG_WARNING("Maximum mean marker fiducial registration error exceeded for tool: " << it->second);
+        continue;
+      }
+
+      // tool is seen with acceptable registration error
+      toolUpdated = true;
+      PlusTransformName toolTransformName(it->second, this->GetToolReferenceFrameName());
+      std::string toolSourceId = toolTransformName.GetTransformName();
+      ToolTimeStampedUpdate(toolSourceId, mit->GetTransformToTracker(), TOOL_OK, this->FrameNumber, unfilteredTimestamp);
+    }
+
+    if (!toolUpdated)
+    {
+      // tool is not seen in this frame
+      vtkSmartPointer<vtkMatrix4x4> emptyTransform = vtkSmartPointer<vtkMatrix4x4>::New();
```

Comment:
You can replace

    vtkSmartPointer<vtkMatrix4x4> emptyTransform = vtkSmartPointer<vtkMatrix4x4>::New();
    emptyTransform->Identity(); 

by

    vtkNew<vtkMatrix4x4> emptyTransform;